tests/test_dataset.py

```diff
@@ -238,6 +239,27 @@ def test_group_by_size_greedy():
     ]
 
 
+def test_group_by_equal_size():
+    obj_list1 = [dict(size=i) for i in range(1, 5)]
+    obj_list2 = [{'size': 70}, {'size': 70}, {'size': 70}, {'size': 70}]
+    obj_list3 = [{'size': 4}, {'size': 1}, {'size': 3}, {'size': 2}]
+
+    groups = _group_by_equal_size(obj_list1, 1)
+    assert groups == [
+        [{'size': 4}, {'size': 3}, {'size': 2}, {'size': 1}]
+    ]
+    groups = _group_by_equal_size(obj_list2, 2, 100)
+    assert groups == [
+        [(70, {'size': 70}), (70, {'size': 70})],
```

Comment:
Why does the first assertion take a list of dictionaries while this one takes a list of tuples? The return types seem to conflict.

Reply:
I understood the desired output for `_group_by_equal_size()` to be a list of tuples. I can change this to a list of dictionaries. Do you prefer the desired output to be a list of dictionaries so it is the same type as the greedy approach? The first assertion is meant to test the behavior in the case where the greedy approach is called within `_group_by_equal_size()`.

Reply:
The return should be the same as `_group_by_equal_size` since it should be a drop in replacement for the use in [Dataset.records()](https://github.com/mozilla/python_moztelemetry/blob/master/moztelemetry/dataset.py#L278-L279). This might crash during runtime since its a different type.

The list of tuples is only needed for the sorting in the heapq methods.